tasks/WebSocketTask.cpp

```diff
@@ -0,0 +1,141 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "WebSocketTask.hpp"
+
+#include <seasocks/PrintfLogger.h>
+#include <seasocks/WebSocket.h>
+#include <seasocks/Server.h>
+
+#include <json/json.h>
+
+#include <string.h>
+#include <iostream>
+
+using namespace controldev;
+using namespace seasocks;
+
+struct controldev::JoystickHandler : WebSocket::Handler {
+    WebSocket *connection = nullptr;
+    WebSocketTask *task = nullptr;
+    void onConnect(WebSocket *socket) override{
+        if (connection != nullptr){
+            connection->close();
+        }
+        Json::Value msg = true;
+        socket->send(msg.asString());
+        connection = socket;
+    }
+    void onData(WebSocket *socket, const char *data) override{
+        Json::Value msg;
+        msg = task->updateRawCommand(data);
+        connection->send(msg.asString());
+        if (msg.asBool() && task->controlling)
+            task->_raw_command.write(task->raw_cmd_obj);
+    }
+    void onDisconnect(WebSocket *socket) override{
+        connection = nullptr;
+    }
+};
+
+bool WebSocketTask::updateRawCommand(const char *data){
+    Json::Value jdata;
+    Json::CharReaderBuilder rbuilder;
+    std::unique_ptr<Json::CharReader> const reader(rbuilder.newCharReader());
+    std::string errs;
+
+    if (!reader->parse(data, data+std::strlen(data), &jdata, &errs)){
+        std::cout << errs << std::endl;
```

Comment:
We should have a better feedback about possible invalid messages. The component should either transition to an exception state, or you should have a statistics output port that report valid/invalid messages.

I like the second option better, actually, as it would provide us with rate estimation as well.